quickstart/k8s-the-prod-way/architecture.md

```diff
@@ -0,0 +1,91 @@
+---
+title: A Prod-First Architecture for Kubernetes _Teams_
+---
+
+Infrastructure mis-configuration probably accounts for the most significant subset of serious
+production outages in major systems. Some [reasonable estimates][post-mortems] put the fraction in
+the range of 40-50%.
+
+Many of these failure modes are trivially preventable. The primary objective of this reference
+architecture is to create sensible defaults that make many of these issues either impossible, or
+dramatically less likely.
+
+Our most effective tools in accomplishing this is to **separate resources into loosely-coupled,
+independently-managable sets, based on _risk_ and _functionality_.**
+
+There are many benefits to this approach:
+
+* **Faster release velocity.** Modern infrastructure-as-code tools (_e.g._, Pulumi, Terraform) can
+  take tens of minutes to generate large plans. Splitting up configuration into well-scoped modules
+  is _always_ more time-efficient.
+* **Tractable plans.** It is markedly easier to notice that you're destroying your primary datastore
+  in a 30-line plan than a 900-line plan.
+* **Users get permissions they _need_ rather than global admin permissions.**
+* **Limited blast radius.** Separating (_e.g._) app and storage configration means that the app team
+  can't accidentally destroy the primary datastore.
+* **Independently managable.** For example, the identity stack (containing, _e.g._, GCP IAM) can
+  expose roles and credentials; the permissions of a specific role can be changed without making
+  changes to (say) the application layer.
+
+And so on.
```

Comment:
This is a good example to point out that Pulumi was designed with this sort of fine grained decomposition in mind, and is why we have the concept of a stack. Probably also worth linking to https://pulumi.io/reference/organizing-stacks-projects.html.